### PR TITLE
[image_picker_platform_interface] use XFile instead of PickedFile

### PR DESCRIPTION
**Pull request by balvinderz**
## Description

In this PR,  i have migrated the image_picker_platform_interface to use XFile instead of PickedFile. In further PRs i will migrate the image_picker and image_picker_web to use XFile also. This will allow users to get the length of the file(https://github.com/flutter/flutter/issues/58764) and name of the file(https://github.com/flutter/plugins/pull/2802#issuecomment-642360958) .

## Related Issues

flutter/flutter#58764 flutter/flutter#70886

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [X] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by balvinderz** [comment]
@ditman  Can you please review this

File: packages/image_picker/image_picker_platform_interface/lib/src/platform_interface/image_picker_platform.dart [hunk 1]
```diff
@@ -10,6 +10,8 @@ import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 import 'package:image_picker_platform_interface/src/method_channel/method_channel_image_picker.dart';
 import 'package:image_picker_platform_interface/src/types/types.dart';
 
+import '../../image_picker_platform_interface.dart';
```
Comment (ditman) [inline packages/image_picker/image_picker_platform_interface/lib/src/platform_interface/image_picker_platform.dart:13]:
```suggestion
import 'package:cross_file/cross_file.dart';
```

File: packages/image_picker/image_picker_platform_interface/lib/src/types/picked_file/lost_data.dart [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:cross_file/cross_file.dart';
```
Comment (ditman) [inline packages/image_picker/image_picker_platform_interface/lib/src/types/picked_file/lost_data.dart:5]: Let's put this lost_data outside of the picked_file subdirectory, now that it's not going to use picked_file. Creating a copy of the class in: `lib/src/types/lost_data.dart` is probably enough.

File: packages/image_picker/image_picker_platform_interface/pubspec.yaml [hunk 3]
```diff
@@ -3,14 +3,15 @@ description: A common platform interface for the image_picker plugin.
 homepage: https://github.com/flutter/plugins/tree/master/packages/image_picker/image_picker_platform_interface
 # NOTE: We strongly prefer non-breaking changes, even at the expense of a
 # less-clean API. See https://flutter.dev/go/platform-interface-breaking-changes
-version: 1.1.5
+version: 1.2.0
```
Comment (ditman) [inline packages/image_picker/image_picker_platform_interface/pubspec.yaml:6]: This is a backwards incompatible change, this must be version `2.0.0`

File: packages/image_picker/image_picker_platform_interface/pubspec.yaml [hunk 4]
```diff
@@ -3,14 +3,15 @@ description: A common platform interface for the image_picker plugin.
...
 
 dependencies:
   flutter:
     sdk: flutter
   meta: ^1.1.8
   http: ^0.12.1
   plugin_platform_interface: ^1.0.2
+  cross_file: ^0.1.0
```
Comment (ditman) [inline packages/image_picker/image_picker_platform_interface/pubspec.yaml:14]: We just published `^0.2.0`, please use that one! :)

File: packages/image_picker/image_picker_platform_interface/test/picked_file_html_test.dart [hunk 5]
```diff
@@ -8,8 +8,8 @@ import 'dart:convert';
 import 'dart:html' as html;
 import 'dart:typed_data';
 
+import 'package:cross_file/cross_file.dart';
```
Comment (ditman) [inline packages/image_picker/image_picker_platform_interface/test/picked_file_html_test.dart:11]: No need to update the picked_file tests to use XFile.

File: packages/image_picker/image_picker_platform_interface/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,10 @@
+## 1.2.0
+
+**breaking changes**:
+- pickImage now returns XFile instead of PickedFile
+- pickVideo now returns XFile instead of PickedFile
+- changed LostData file parameter type to XFile
+
```
Comment (ditman) [inline packages/image_picker/image_picker_platform_interface/CHANGELOG.md:7]: This also needs to remove the PickedFile class from this package, and its tests. That way we only use XFile from this major version on. I'll let the plugins team know tomorrow at standup that this major version change is incoming.

**Review by ditman — COMMENTED** [review]
I'll let people on the team know tomorrow that this change is incoming. Please address the comments below, if you can!

Also, the `format` precheck is not passing because of the following file:

```
These files are not formatted correctly (see diff below):
  image_picker/image_picker_platform_interface/lib/image_picker_platform_interface.dart
```

You should be able to run `dartfmt -w lib/image_picker_platform_interface.dart` to have the dartfmt tool automatically correct the above for you.

**Comment by balvinderz** [comment]
@ditman  can you review this again .

**Comment by ditman** [comment]
@balvinderz this looks good, but I think the team prefers a backwards-compatible version (keeping the old features, and adding new methods for the new features) instead of a major version bump. I'll verify this tomorrow, and if so, I'll do the necessary changes in this PR. Thank you very much for the contribution!

**Comment by ditman** [comment]
@balvinderz migrating the platform_interface package to null-safety (in a backwards-compatible way) has taken a higher priority than landing this feature, so the team has decided to work on that first.

**Thank you very much for your effort with this PR, but we can't work on merging this PR right now**. I'll close this PR, and once the null safety migration is complete, we can revisit this, or maybe re-do it in the new null-safe code. Otherwise, the PR will take too much work to keep rebased.

**Pull request closed without merging**